aries/aries_vcx/src/protocols/did_exchange/mod.rs

```diff
@@ -103,11 +106,25 @@ pub async fn resolve_enc_key_from_invitation(
                 "resolve_enc_key_from_invitation >> Resolved did document {}",
                 output.did_document
             );
-            let key = resolve_first_key_agreement(&output.did_document)?;
```

Comment:
Do you think `resolve_first_key_agreement()` is still a valid/useful function with this change? Just curious, no strong opinion from my side

Reply:
hmm yea it's a tough one. Ideally, DIDDocs that we resolve will have:
* a service with a recipient key

and then we just just use that recipient key as the peer's "encryption key".

However it is possible that some DIDs may not be perfect, and they will have a service, but no recipient keys, in which case, i think it's a decent backup solution to use the first key agreement key (`resolve_first_key_agreement`).

The above is the `resolve_enc_key_from_did_doc` after these changes. I don't think it's perfect, but i think this might be a situation where we need to handle random interop edge cases as they reveal themselves over time... atleast we know it interops with ourselves and with ACApy